source/common/crypto/utility.h

```diff
@@ -6,11 +6,26 @@
 #include "envoy/buffer/buffer.h"
 
 #include "absl/strings/string_view.h"
+#include "openssl/evp.h"
 
 namespace Envoy {
 namespace Common {
 namespace Crypto {
 
+struct VerificationOutput {
+  /**
+   * Verification result. If result_ is true, error_message_ is empty.
+   */
+  bool result_;
+
+  /**
+   * Error message when verification failed.
+   */
+  std::string error_message_;
```

Comment:
Maybe leave a TODO for switching to `absl::StatusOr` if/when that becomes available.